Segmentation Violation using DeploymentSettings in v0.22.0

### What happened?

While doing a pulumi up I received this segmentation violation:

```
panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x91ea40]
    goroutine 69 [running]:
    github.com/pulumi/pulumi-pulumiservice/provider/pkg/provider.(*PulumiServiceDeploymentSettingsInput).ToPropertyMap(0x40000f74b8, 0x40000f7520, 0x0, 0x20?)
        /home/runner/work/pulumi-pulumiservice/pulumi-pulumiservice/provider/pkg/provider/deployment_settings.go:87 +0x660
    github.com/pulumi/pulumi-pulumiservice/provider/pkg/provider.(*PulumiServiceDeploymentSettingsResource).Create(0x4000685240, 0x4000110b90?)
        /home/runner/work/pulumi-pulumiservice/pulumi-pulumiservice/provider/pkg/provider/deployment_settings.go:776 +0x204
    github.com/pulumi/pulumi-pulumiservice/provider/pkg/provider.(*pulumiserviceProvider).Create(0x400010ce40, {0x6dfb54?, 0x40?}, 0x9f6f80?)
        /home/runner/work/pulumi-pulumiservice/pulumi-pulumiservice/provider/pkg/provider/provider.go:225 +0xe4
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler.func1({0xdacc98, 0x40006b8ea0}, {0xb27d60?, 0x4000110b90})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.115.3-0.20240507143413-cffdfd1fa489/proto/go/provider_grpc.pb.go:593 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0xdacc98, 0x40006b8720}, {0xb27d60, 0x4000110b90}, 0x400036e3e0, 0x400010e750)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x2e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler({0xb86020?, 0x400010ce40}, {0xdacc98, 0x40006b8720}, 0x4000342500, 0x400036f4a0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.115.3-0.20240507143413-cffdfd1fa489/proto/go/provider_grpc.pb.go:595 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x400035e800, {0xdacc98, 0x40006b8690}, {0xdb5940, 0x40002ce780}, 0x4000303320, 0x400018faa0, 0x1566040, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1369 +0xba0
    google.golang.org/grpc.(*Server).handleStream(0x400035e800, {0xdb5940, 0x40002ce780}, 0x4000303320)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1780 +0xc80
    google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
    created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 85
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1030 +0x150

  pulumiservice:index:DeploymentSettings (deployment_settings):
    error: error reading from server: EOF
```

### Example

Use the TS example from

https://github.com/pulumi/pulumi-pulumiservice/tree/main/examples/ts-deployment-settings

With versions 0.22.0 and 0.22.1 you'll get a SIGSEGV.

Works well with version 0.21.4

### Output of `pulumi about`

at 14:57:14 ❯ pulumi about
CLI          
Version      3.122.0
Go Version   go1.22.4
Go Compiler  gc

Plugins
KIND      NAME           VERSION
language  nodejs         unknown
resource  pulumiservice  0.21.4

Host     
OS       debian
Version  12.6
Arch     aarch64

This project is written in nodejs: executable='/usr/local/bin/node' version='v20.15.0'

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/or
User           
Organizations  
Token type     personal


### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -1,6 +1,8 @@
 ### Improvements
 
 ### Bug Fixes
+
 - Fixed import by refactoring Read method of EnvironmentVersionTag resource [311](https://github.com/pulumi/pulumi-pulumiservice/issues/311)
+- Fix a panic in DeploymentSettings [#356](https://github.com/pulumi/pulumi-pulumiservice/pull/356)
 
 ### Miscellaneous

--- a/examples/go-environments/go.mod
+++ b/examples/go-environments/go.mod
@@ -44,7 +44,7 @@ require (
 
 require (
 	github.com/Microsoft/go-winio v0.6.1 // indirect
-	github.com/ProtonMail/go-crypto v0.0.0-20230828082145-3c4c8a2d2371 // indirect
+	github.com/ProtonMail/go-crypto v1.0.0 // indirect
 	github.com/aead/chacha20 v0.0.0-20180709150244-8b13a72661da // indirect
 	github.com/blang/semver v3.5.1+incompatible // indirect
 	github.com/cheggaaa/pb v1.0.29 // indirect
@@ -53,7 +53,7 @@ require (
 	github.com/emirpasic/gods v1.18.1 // indirect
 	github.com/go-git/gcfg v1.5.1-0.20230307220236-3a3c6141e376 // indirect
 	github.com/go-git/go-billy/v5 v5.5.0 // indirect
-	github.com/go-git/go-git/v5 v5.11.0 // indirect
+	github.com/go-git/go-git/v5 v5.12.0 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/golang/glog v1.1.2 // indirect
 	github.com/grpc-ecosystem/grpc-opentracing v0.0.0-20180507213350-8e809c8a8645 // indirect
@@ -73,8 +73,8 @@ require (
 	github.com/rogpeppe/go-internal v1.11.0 // indirect
 	github.com/sabhiram/go-gitignore v0.0.0-20210923224102-525f6e181f06 // indirect
 	github.com/santhosh-tekuri/jsonschema/v5 v5.0.0 // indirect
-	github.com/sergi/go-diff v1.3.1 // indirect
-	github.com/skeema/knownhosts v1.2.1 // indirect
+	github.com/sergi/go-diff v1.3.2-0.20230802210424-5b0b94c5c0d3 // indirect
+	github.com/skeema/knownhosts v1.2.2 // indirect
 	github.com/spf13/cobra v1.8.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/texttheater/golang-levenshtein v1.0.1 // indirect

--- a/provider/pkg/provider/deployment_settings.go
+++ b/provider/pkg/provider/deployment_settings.go
@@ -85,7 +85,7 @@ func (ds *PulumiServiceDeploymentSettingsInput) ToPropertyMap(plaintextInputSett
 							importSecretValue(sshAuthPropertyMap, "sshPrivateKey", ds.SourceContext.Git.GitAuth.SSHAuth.SSHPrivateKey, isInput)
 						}
 					}
-					if ds.SourceContext.Git.GitAuth.SSHAuth.Password.Value != "" {
+					if ds.SourceContext.Git.GitAuth.SSHAuth.Password != nil && ds.SourceContext.Git.GitAuth.SSHAuth.Password.Value != "" {
 						if mergeMode {
 							var plaintextValue *pulumiapi.SecretValue
 							var currentCipherValue *pulumiapi.SecretValue

